src/components/WeeklyCalendar/helpers/getFormattedWeek.js

```diff
@@ -0,0 +1,22 @@
+export default function getFormattedWeek(firstDay, locale) {
+    const lastDay = new Date(firstDay);
+    lastDay.setDate(firstDay.getDate() + 6);
+
+    let formattedWeek = `${new Intl.DateTimeFormat(locale, {
```

Comment:
It will be a good idea to cache the Intl instances here since they are the same always and always
```
const formatMonth= new Intl.DateTimeFormat(locale, { month: 'short' })
formatMonth(lastDay) 
```
note: by caching I meant put the instance creation out of the function(singleton pattern)